### ISSUE TITLE
Question: Google+ sunsetting effects?

### ISSUE THREAD

**Issue by TuulaP** [spring-social/spring-social-google#115]
Hi, and apologies for disturbing so early in the year. Tried to search existing issues and documentation of this one, but is it so that as [Google+ is going away](https://developers.google.com/+/integrations-shutdown), this library would need some updates to migrate to Google Sign-in? Is there some plans for that?  Or do we need to find alternatives? Any recommendations?

**Comment by amontenegro** [comment]
I have the same question, at least for google signin, is the google+ sunset going to affect google sign in?

**Comment by PawelJagus** [comment]
The same question here. Will there be any update to support migration from Google Plus SignIn to Google SignIn?

**Comment by felix-zaslavskiy** [comment]
The People API seems to return Work history etc.., Seems like a valuable update to get that code released.

**Comment by n1zea144** [comment]
@amontenegro I have the same concern.  A cursory look through the code leads me to believe that the sign-on functionality is not handled via OAuth or plus.me api/scopes, but rather OAuth2: [GoogleOAuth2Template](https://github.com/spring-social/spring-social-google/blob/master/src/main/java/org/springframework/social/google/connect/GoogleOAuth2Template.java).

Having said that, I'm not certain that sign-on will be unaffected by the Google+ API shutdown.  It would be nice to get a confirmation from someone on the dev team.

**Comment by amontenegro** [comment]
> @amontenegro I have the same concern. A cursory look through the code leads me to believe that the sign-on functionality is not handled via OAuth or plus.me api/scopes, but rather OAuth2: [GoogleOAuth2Template](https://github.com/spring-social/spring-social-google/blob/master/src/main/java/org/springframework/social/google/connect/GoogleOAuth2Template.java). For what its worth, I just disabled the Google+ API in our projects Google API dashboard and sign-on is still working.
> 
> Having said that, I'm not certain that sign-on will be unaffected by the Google+ API shutdown. It would be nice to get a confirmation from someone on the dev team.

Hi @n1zea144, you are right, sign in was handled by oauth2, however, profile info was fetched from google+ endpoints, so, at the end, I extended the `GoogleTemplate` so I fetch the user info (name + email) from `https://www.googleapis.com/oauth2/v2/userinfo` using OAuth2Operations.getUserinfo();

You can take a look at this PR:

https://github.com/ORCID/ORCID-Source/pull/5187/files#diff-23ff651cfe8bb56c3f5d74e4baf86f41R7

**Comment by n1zea144** [comment]
Thanks @amontenegro for the patch!  I haven't checked, but did you attempt to submit a PR back to the primary spring-social-google repos (of course we can always merge these changes directly from your ORCID repos).

**Comment by PawelJagus** [comment]
> For what its worth, I just disabled the Google+ API in our projects Google API dashboard and sign-on is still working.

Yeah... I did the same and it was working for... one day more. Better to check your app on a daily basis because they announced they would switch it off step by step up to March the 7th.

**Comment by felix-zaslavskiy** [comment]
I created pull request to update the People API so same data can be retrieved that was previously done via Google + API. Hope some maintainer notices.

**Comment by michaelwiles** [comment]
Anyone got a github repo with this change working that I can pull and build my own version? Don't see it being merged

**Comment by mjacksonlgh** [comment]
just fyi - I disabled Google+ for our spring-social-google 1.0.0 app and the OAuth did indeed fail.

**Comment by oyvindhorneland** [comment]
With 1.0.0 I verified that signin failed after I removed the Google Plus API. A simple upgrade to the already released version 1.1.3 (https://mvnrepository.com/artifact/com.github.spring-social/spring-social-google/1.1.3) fixed the signin part.

(Not sure how we missed the 1.1.3 upgrade previously, perhaps due to the group change described in the project README file. Anyway 1.1.3 can sign in without Google Plus API.)
